### ISSUE TITLE
Local Capture and Review -> Optional Remote Upload

### ISSUE THREAD

**Issue by brianfinley** [ConradIrwin/showterm.io#16]
I love this tool, and am planning to use it for all kinds of demos, training, and process documentation.  But, I need to be able to run it on local machines.

Ideally, I could have it installed on the notebooks of many field staff (mostly linux), where the sessions could be captured and reviewed locally.  After review, sessions could be uploaded to either a company server or the showterm.io site.

Architecturally I see this as pretty simple:
  1) The local machine runs it's own private server, that let's the user locally review the sessions (http://localhost:3000/).
  2) For any given session, the user can choose to upload it to an alternate network connected server.

Command wise, it could look like this:
  showterm --list
    # shows a listing of sessions on $SHOWTERM_SERVER

  showterm --name  "New Name" http://localhost:3000/9d4f95fade1c52ad86837
    # adds "New Name" as an attribute of 9d4f95fade1c52ad86837 that appears in the --list

  showterm --copy http://localhost:3000/9d4f95fade1c52ad86837  --server http://my.corp:3000/
    # copies the session from  up to --server

Lastly, having a simple script to setup the local instance.  Ideally using something like sqlite with full setup by the script, and no expectation for conflict with other apps on the system.

Thanks for creating this great tool!  Keep it up. :-)

**Comment by alingse** [comment]
@ConradIrwin 
Local Capture might be useful when record some bussiness code to the remote co-workers

🤔

**Comment by kousu** [comment]
I've been using showterm a lot this pandemic year for exactly @alingse's use case. I feel a little bad leaning my paid work so heavily on it, too, but it's just so much more useful and tightly done than Zoom or https://asciinema.org/ or using the mac or gnome screenrecorders.

I also want more control over the files though. Personally, I don't need `showterm --list`, but I would like to be able to store session as regular files, say `*.scriptreplay`? `*.showterm` (themselves just zip files)?,

- saving a .zip of the script and timing files before or instead of uploading (so that they can be archived and reviewed by `scriptreplay`)
- prompting about if we want to upload the file or not
- uploading after-the-fact (`showterm upload ...`)
- downloading that .zip from the website
    - I can extract it from the javascript but my teammates can't; it'd be nicer if this was just built-in.

**Comment by ConradIrwin** [comment]
@kousu I'd absolutely accept changes so that downloading a zip works (by just adding `.zip` to the end of the URL?)

I'd prefer to not prompt by default (but happy to have local saving as an option so you could do `showterm -o tutorial-3.zip` or similar, then `showterm --upload tutorial-3.zip`).

(P.S. One problem I ran into with your previous commit is that `showterm.io` is on an unsupported heroku stack, and upgrading that requires upgrading (at least) the version of ruby (but probably also more things..). So while I'd love to deploy a new version, we need to do some work and testing to make that happen. (Maybe time to rewrite in go? :D))

**Comment by kousu** [comment]
You're right, `-o` is better than prompting. And `.zip` on the end of the url is super, that's genius.

(It's only been, what, a decade? Enough time for a language to fade away I guess :P)
